src/screens/home/hooks/useLauncherWrapper.ts

```diff
@@ -0,0 +1,37 @@
+import CozyClient from 'cozy-client'
+
+import { LauncherContext } from '/libs/connectors/models'
+import { useLauncherClient } from '/screens/home/hooks/useLauncherClient'
+
+const isLauncherReady = (
+  context: LauncherContext
+): context is LauncherContext => context.state !== 'default'
+
+const isClientReady = (client: CozyClient | undefined): client is CozyClient =>
+  Boolean(client)
+
+export const useLauncherWrapper = (
+  launcherContext: LauncherContext
+): {
+  canDisplayLauncher: () => boolean
+  launcherClient?: CozyClient
+} => {
+  const { launcherClient } = useLauncherClient(launcherContext.value)
+
+  const canDisplayLauncher = (): boolean => {
+    if (!isLauncherReady(launcherContext)) {
+      return false
+    }
+
+    if (!isClientReady(launcherClient)) {
+      return false
+    }
+
+    return true
+  }
```

Comment:
nit: can be simplified as
```js
const canDisplayLauncher = (): boolean =>
  isLauncherReady(launcherContext) && isClientReady(launcherClient)
```

With that form, I'm wondering if we should still have an entire hook for this as we can just have those two lines in `useLauncherContext` so we won't have to understand the difference and relationship between the `Context` and the `Wrapper`

```js
const { launcherClient } = useLauncherClient(launcherContext.value)
const canDisplayLauncher = (): boolean =>
  isLauncherReady(launcherContext) && isClientReady(launcherClient)
```

But this is very subjective, I know it it easier for you to read split logic contrary to me who prefer condensed logic.

Reply:
Agree, I will handle a refactor into another PR, I have to think about a pattern